Sources/NotificationApi_Ios_Sdk/AppDelegate/NotificationApiAppDelegate.swift

```diff
@@ -0,0 +1,53 @@
+import UIKit
+
+@available(iOSApplicationExtension, unavailable)
+open class NotificationApiAppDelegate: NSObject, UIApplicationDelegate, UNUserNotificationCenterDelegate {
+    // MARK: - Init
+
+    override init() {
+        super.init()
+
+        UNUserNotificationCenter.current().delegate = self
+        UIApplication.shared.registerForRemoteNotifications()
+    }
+
+    // MARK: - Delegate Functions
+
+    public func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
+        Task {
+            do {
+                let token = deviceToken.toString()
+
+                notificationApi(apnTokenDidChange: token)
+
+                try await NotificationApi.shared.syncApn(token: token)
+            } catch {
+                print("NotificationApi error: \(error)")
+            }
+        }
+    }
+
+    public func application(_ application: UIApplication, didFailToRegisterForRemoteNotificationsWithError error: Error) {
+        print("NotificationAPI error: Failed to register for remote notifications: \(error.localizedDescription)")
+    }
+
+    public func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse) async {
```

Comment:
This function will handle if a notification is clicked in the background. Again, this will call a hook so users can do things with this. Note that this doesn't have an actual implementation yet